src/role.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+from enum import Enum, auto
+from flask_login import current_user
+from flask import redirect, flash
+from functools import wraps
+
+class Role(Enum):
+    READER = auto()
+    WRITER = auto()
+
+    @staticmethod
+    def for_user(user_id: str) -> Role:
+        if user_id == 'TheMattBarnfield' or user_id == "TestUser":
+            return Role.WRITER
+        else:
+            return Role.READER
+
+
+def require_writer(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        if current_user.role == Role.WRITER:
+            return f(*args, **kwargs)
```

Comment:
You could wrap f with login_required as well so that you don't need to decorate each route with both login_required and require_writer.